### PR TITLE
Some fixes: 

### PR DESCRIPTION
**Pull request by luantak**
- build a string instead of printing, to allow for more flexible usage

- export all Struct Fields/ Types to make everything user accessible,
this also enables to encode the data to formats like JSON

- remove the struct definition from the README
(Users should use the automatically generated documentation instead,
because it cannot get outdated)

- GetVersion returns by value to indicate, that it can't be nil

- simplify code

### CODE REVIEW

File: osinfo_test.go [hunk 1]
```diff
@@ -7,14 +7,14 @@ import (
...
-	fmt.Println(x.Name)
-	fmt.Println(x.Version)
+	fmt.Println(info.Arch)
+	fmt.Println(info.Runtime)
+	fmt.Println(info.Name)
+	fmt.Println(info.Version)
 	fmt.Println("-------------")
-	x.PrintInfo()
+
+	fmt.Println(info)
```
Comment (luantak) [inline osinfo_test.go:19]: This automatically uses the declared String() method of the release type.

File: ver_linux.go [hunk 2]
```diff
@@ -20,98 +18,89 @@ import (
...
-	return true
-}
 
-func readFile(filename string) string {
-	content, err := ioutil.ReadFile(filename)
-	if err != nil {
-		return ""
+	if verf := strings.Split(verf, "="); len(verf) >= 2 {
+		info.Version = verf[1]
+	}
```
Comment (luantak) [inline ver_linux.go]: Add length checks to not panic

File: ver_linux.go [hunk 3]
```diff
@@ -20,98 +18,89 @@ import (
...
-			inf.nix.Distro = "alpine"
-			inf.nix.PkgMng = "apk"
-
-		} else {
-			inf.nix.Distro = "unknown"
-			inf.nix.PkgMng = "unknown"
-		}
+	out, err := exec.Command("uname", "-r").Output()
+	if err != nil {
+		return info
```
Comment (JustinTimperio) [inline ver_linux.go]: If it can't get the kernel version, it should still continue on and try to read `/etc/os-release`

Comment (luantak) [reply ver_linux.go]: ok

File: osinfo.go [hunk 4]
```diff
@@ -10,51 +14,68 @@ type Release struct {
...
-	bsd     bsdRelease
-	osx     darwinRelease
+	Windows WindowsRelease
+	Linux   LinuxRelease
+	BSD     BsdRelease
+	MacOS   DarwinRelease
 }
 
-type windowsRelease struct {
+type WindowsRelease struct {
```
Comment (JustinTimperio) [inline osinfo.go]: Why export this?

File: osinfo.go [hunk 5]
```diff
@@ -10,51 +14,68 @@ type Release struct {
...
 	Build string
 }
 
-type linuxRelease struct {
-	Distro string
-	Kernel string
-	PkgMng string
+type LinuxRelease struct {
```
Comment (JustinTimperio) [inline osinfo.go]: Why export this?

File: osinfo.go [hunk 6]
```diff
@@ -10,51 +14,68 @@ type Release struct {
...
+	Distro     string
+	Kernel     string
+	PkgManager string
 }
 
-type bsdRelease struct {
-	Kernel string
-	PkgMng string
+type BsdRelease struct {
```
Comment (JustinTimperio) [inline osinfo.go]: Why export this?

File: osinfo.go [hunk 7]
```diff
@@ -10,51 +14,68 @@ type Release struct {
...
+	Kernel     string
+	PkgManager string
 }
 
-type darwinRelease struct {
+type DarwinRelease struct {
```
Comment (JustinTimperio) [inline osinfo.go]: Why export this?

File: ver_windows.go [hunk 8]
```diff
@@ -40,16 +42,14 @@ func GetVersion() *Release {
 	inf.Name = pname
```
Comment (JustinTimperio) [inline ver_windows.go]: You missed lines 38-42. That's a really nice simplification but we should keep it consistent

File: osinfo.go [hunk 9]
```diff
@@ -10,51 +14,68 @@ type Release struct {
...
-	} else if i.Runtime == "linux" {
-		fmt.Println("Kernel:", i.nix.Kernel)
-		fmt.Println("Distro:", i.nix.Distro)
-		fmt.Println("Package Manager:", i.nix.PkgMng)
+	return b.String()
+}
 
-	} else if i.Runtime == "darwin" {
-		fmt.Println("Kernel:", i.osx.Kernel)
+func cleanString(s string) string {
```
Comment (JustinTimperio) [inline osinfo.go:66]: This was smart

File: ver_windows.go (hunk 8 above)
Comment (JustinTimperio) [reply ver_windows.go]: sorry think you got this and github is just slow

**Comment by JustinTimperio** [comment]
Great work here @lu4p, thanks for the PR

**Pull request merged**